src/System.Management.Automation/engine/ArgumentTypeConverterAttribute.cs

```diff
@@ -147,15 +147,14 @@ internal object Transform(EngineIntrinsics engineIntrinsics, object inputData, b
                     // Note - this is duplicated in ExecutionContext.cs as parameter binding for script cmdlets can avoid this code path.
```

Comment:
@KirkMunro, your last commit had 15 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Test-Connection.Ping.Default parameter set is 'Ping'
```powershell
The value Ignore is not supported for an ActionPreference variable. The provided value should be used only as a value for a preference parameter, and has been replaced by the default value. For more information, see the Help topic, "about_Preference_Variables."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Test-Connection.Tests.ps1: line 40
40:             $result = Test-Connection $targetName

```
Test-Connection.Ping.Count parameter
```powershell
The value Ignore is not supported for an ActionPreference variable. The provided value should be used only as a value for a preference parameter, and has been replaced by the default value. For more information, see the Help topic, "about_Preference_Variables."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Test-Connection.Tests.ps1: line 60
60:             $result1 = Test-Connection -Ping $targetName -Count 1

```
Test-Connection.Ping.Quiet works
```powershell
The value Ignore is not supported for an ActionPreference variable. The provided value should be used only as a value for a preference parameter, and has been replaced by the default value. For more information, see the Help topic, "about_Preference_Variables."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Test-Connection.Tests.ps1: line 68
68:             $result1 = Test-Connection $targetName -Count 1 -Quiet

```
Test-Connection.Ping.Force IPv4 with implicit PingOptions
```powershell
The value Ignore is not supported for an ActionPreference variable. The provided value should be used only as a value for a preference parameter, and has been replaced by the default value. For more information, see the Help topic, "about_Preference_Variables."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Test-Connection.Tests.ps1: line 89
89:             $result = Test-Connection $hostName -Count 1 -IPv4

```
Test-Connection.Ping.Force IPv4 with explicit PingOptions
```powershell
The value Ignore is not supported for an ActionPreference variable. The provided value should be used only as a value for a preference parameter, and has been replaced by the default value. For more information, see the Help topic, "about_Preference_Variables."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Test-Connection.Tests.ps1: line 100
100:             $result1 = Test-Connection $hostName -Count 1 -IPv4 -MaxHops 10 -DontFragment

```